server/player/player.go

```diff
@@ -2028,8 +2028,54 @@ func (p *Player) CollectExperience(value int) bool {
 	if time.Since(p.lastXPPickup.Load()) < time.Millisecond*100 {
 		return false
 	}
+	value = p.handleMending(value)
 	p.lastXPPickup.Store(time.Now())
-	return p.AddExperience(value) > 0
+	if value > 0 {
+		return p.AddExperience(value) > 0
+	}
+
+	p.PlaySound(sound.Experience{})
+	return true
+}
+
+// handleMending handles the mending enchantment when collecting experience, it then returns the leftover experience.
+func (p *Player) handleMending(value int) int {
```

Comment:
Would probably name this method `mendItems(xp int)` to be a bit more descriptive.